rfd/0036/README.md

```diff
@@ -13,281 +13,800 @@ state: draft
+
+
+
+### Auto Scaling Groups
+
+AWS' [Auto Scaling Groups](https://aws.amazon.com/autoscaling/) (ASGs) represent a very basic example of service definition with supervision. Though most people think of ASGs as a way to respond to performance metrics, the first mode of operation is actually just as a supervisor with health checks:
+
+> Whether you are running one Amazon EC2 instance or thousands, you can use Auto Scaling to detect impaired Amazon EC2 instances and unhealthy applications, and replace the instances without your intervention. This ensures that your application is getting the compute capacity that you expect.
+
+Once an Auto Scaling Group is defined (a service definition, as described elsewhere in this document), AWS customers can easily change the number of running instances of that service. It's [entirely optional to connect performance metrics to it](https://aws.amazon.com/autoscaling/getting-started/) for fully automated scaling.
+
+CloudFormation and Auto Scaling Groups represent some of the most important baseline features proposed for Mariposa, especially the deep integration with the cloud IaaS, but neither of them supports the management features that are also required for users building and running cloud applications today.
+
+
+
+### Compose
+
+Most early Docker users experienced a moment of great joy when discovering Docker Compose and using it to run a composed set of containers via a single manifest. Sadly, that joy was often later followed by the realization that running Docker Compose in production across multiple VMs required substantially more work, including the use of Docker Swarm (or Swarm Mode on more recent versions of Docker). On Triton, however, because the Docker API is exposed via sdc-docker and users can address an entire data center without needing Docker Swarm or creating any VMs, Docker Compose has proven incredibly powerful.
+
+The critical feature Docker Compose offers over the Docker client itself is the ability to define [one or more services in a manifest, typically named `docker-compose.yaml`](https://docs.docker.com/compose/compose-file/). The [service definition](https://docs.docker.com/compose/compose-file/#compose-file-structure-and-examples) includes features that are common to other examples here, including Kubernetes, Mesos+Marathon, and others. When combined with the [Autopilot Pattern](http://autopilotpattern.io), it is possible to [automate the operations of stateful applications](https://www.joyent.com/blog/persistent-storage-patterns#databases) so that [they operate statelessly from the perspective of the scheduler](https://www.joyent.com/blog/app-centric-micro-orchestration#handling-persistent-data). We've even demonstrated this with [MySQL on Autopilot](https://github.com/autopilotpattern/mysql). For example:
```

Comment:
As mentioned above, one of the immediate value props I noticed with Dockerfiles and Compose was the ability to have infrastructure defined as code and versioned and self-documenting. This comment may be beyond the scope of your paragraph.